### ISSUE TITLE
PolyRatFun(expand) does not expand substituted expressions

### ISSUE THREAD

**Issue by tueda** [vermaseren/form#56]
With the following code, the expression H is printed with a non-zero result:
```
CF rat;
S x;
PolyRatFun rat;
L F = rat(1,1+x);
L G = rat(1-x,1);
.sort
PolyRatFun rat(expand,x,1);
Drop;
L H = F - G;
*.sort;  * <-- (1)
P;
.end
```
```
   H =
      rat(1 - x) + rat(1 - x)*rat(-1,1);
```
Enabling an extra `.sort` at line (1) gives the correct result:
```
   H = 0;
```

**Comment by tueda** [comment]
Another thing that differs from my expectation is, when I put `PolyRatFun` at line (2):
```
CF rat;
S x;
PolyRatFun rat;
L F = rat(1,1+x);
L G = rat(1-x,1);
.sort
PolyRatFun rat(expand,x,1);
Drop;
L H = F - G;
.sort;  * <-- (1)
PolyRatFun;  * <-- (2)
P;
.end
```
then it gives
```
   H =
      rat(1 - x) + rat(-1,1)*rat(1 - x);
```
Sounds like `PolyRatFun rat(expand)` is ignored at `.sort` of line (1).

**Comment by vermaseren** [comment]
No it is not. Otherwise it would not have expanded the denominator.
More likely there is somewhere a place that adds the rat(-1,1) because the expand mode
did not overwrite that part.

Jos

**Comment by tueda** [comment]
@vermaseren OK. You are right. Adding another `.sort` gave me the expected result in this case:
```
CF rat;
S x;
PolyRatFun rat;
L F = rat(1,1+x);
L G = rat(1-x,1);
.sort
PolyRatFun rat(expand,x,2);  * <-- up to x^2
Drop;
L H = F - G;
.sort;  * <-- (1)
.sort;  * <-- (3)
PolyRatFun;  * <-- (2)
P;
.end
```

**Comment by tueda** [comment]
It seems that this issue has the same origin as #111 and fixed in 789edc810a5c8afb3e1d857a32910b796431f069.
